blocks/api/serializer.js

```diff
@@ -93,9 +93,11 @@ export function getCommentAttributes( allAttributes, attributesFromContent ) {
 			const contentValue = attributesFromContent[ key ];
 
 			// save only if attribute if not inferred from the content and if valued
-			return ! ( contentValue !== undefined || allValue === undefined )
-				? Object.assign( toSave, { [ key ]: allValue } )
-				: toSave;
+			if ( ! isNil( contentValue ) || isNil( allValue ) ) {
+				return toSave;
+			}
+
+			return Object.assign( toSave, { [ key ]: allValue } );
```

Comment:
Regarding striping out `null`, I find it a bit strange to define as `undefined` and it doesn't seem to play well with React. Also, they strip out `null` as well.